packages/components/src/icon-button/index.js

```diff
@@ -42,7 +43,7 @@ class IconButton extends Component {
 
 		let element = (
 			<Button aria-label={ label } { ...additionalProps } className={ classes }>
-				{ isString( icon ) ? <Dashicon icon={ icon } /> : icon }
+				{ isString( icon ) ? <Dashicon icon={ icon } ariaPressed={ ariaPressed } /> : icon }
```

Comment:
Not sure I understand this change? why do we need aria-pressed here since we're applying it to the parent button?

Reply:
The underlying issue is that we need to find a way to set the color of a `Dashicon`. Since the web has CSS it can use class names, inheritance, and `fill: currentColor` to get to that. But we don't have that on native. 

Ideally we'd be able to set `color` or `style` on a Dashicon, I took an alternate approach in #12403, but that was easier since I was using Dashicon directly. For this we'd have to make everything in the hierarchy forward the style down to Dashicon. As mentioned on that PR, you can set `extraProps` in `IconButton`, but those will go to the `Button`, not the icon.

We just found something that worked to keep us moving although we were aware it wasn't great, so I'm happy to discuss a better solution 😁

Reply:
I'm concerned about these small tweaks having an impact on the web code base, can't we create an `icon-button` native version instead?

Reply:
> The underlying issue is that we need to find a way to set the color of a `Dashicon`. Since the web has CSS it can use class names, inheritance, and `fill: currentColor` to get to that. But we don't have that on native.

@koke Can you point me to where this occurs? As best I can follow, the main impact of the `ariaPressed` prop is in changing the class behavior of `Dashicon` to apply `dashicon-active`, but I see no reference to `dashicon-active` anywhere in Gutenberg or [Gutenberg Mobile](https://github.com/wordpress-mobile/gutenberg-mobile) aside from where we apply it.

I'd tend to agree with @youknowriad here. "Pressed" does not resound to me as a characteristic of an icon, and thus it does not make sense as a prop. It _does_, however, seem natural to describe the state of an `IconButton`.

Reply:
I think this is what's setting the style https://github.com/WordPress/gutenberg/blob/master/packages/components/src/primitives/svg/style.native.scss#L6

I agree that this isn't ideal. It seems like the introduction of `iconProps` in #13977 could be a good solution to pass custom styles from `IconButton` to `Dashicon`?

Reply:
I don't know that #13977 is an improvement here, as I expect you'd continue to treat it as a _pressed icon_, which is the root of the issue ("pressed" is not a sensible state of an icon).

Reply:
Ah, that's right. 
It looks like the right logic here would be that `IconButton` set different style props on `Dashicon` depending on `isActive`. This seems to be what the web is doing, although [via CSS](https://github.com/WordPress/gutenberg/blob/master/packages/components/src/icon-button/style.scss#L32).
However this is [overriden by CSS](https://github.com/WordPress/gutenberg/blob/master/packages/components/src/toolbar-button/style.scss#L61) in the `ToolbarButton` component.

We keep bumping into this on native: the web can rely on nested CSS definition to pass style down to lower level components, but we can't do that unless we pass them explicitly. I still think `ToolbarButton` should be somehow deciding what the style for `Dashicon` is when normal/pressed, which is why now having `iconProps` in `IconButton` seems like a way out.

@etoledom can you handle this (removing ariaPressed from Dashicon) as part of #13977?